pkg/network/go/bininspect/symbols.go

```diff
@@ -0,0 +1,297 @@
+		return nil, elf.ErrNoSymbols
+	}
+
+	// Checking the symbol section size is aligned to a multiplication of Sym64Size.
+	if symbolSection.Size%elf.Sym32Size != 0 {
+		return nil, errors.New("length of symbol section is not a multiple of Sym32Size")
+	}
+
+	// Checking the symbol section link is valid.
+	if symbolSection.Link <= 0 || symbolSection.Link >= uint32(len(f.Sections)) {
+		return nil, errors.New("section has invalid string table link")
+	}
+
+	// Allocating entries for all wanted symbols.
+	symbols := make([]elf.Symbol, 0, len(wantedSymbols))
+
+	// Copying the wanted symbol set, so we can modify it during runtime.
+	copyOfWantedSymbols := wantedSymbols.Clone()
+	// Extracting the min and max symbol length.
+	minSymbolNameSize, maxSymbolNameSize := getSymbolBoundaries(copyOfWantedSymbols)
+	// Pre-allocating a buffer to read the symbol string into.
+	// The size of the buffer is maxSymbolNameSize + 1, for null termination.
+	symbolNameBuf := make([]byte, maxSymbolNameSize+1)
+	// Pre allocating a buffer for reading the symbol entry in the string table.
+	allocatedBufferForSymbolNameRead := make([]byte, 4)
+	// Pre allocating a buffer for reading the rest of the symbol fields from the symbol section.
+	allocatedBufferForSymbolRead := make([]byte, elf.Sym32Size-len(allocatedBufferForSymbolNameRead))
+
+	// Iterating through the symbol table. We skip the first elf.Sym32Size bytes as they are zeros.
+	for readLocation := int64(elf.Sym32Size); uint64(readLocation) < symbolSection.Size; readLocation += elf.Sym32Size {
+		// Reading the symbol entry in the string table.
+		stringEntry, ok := readSymbolEntryInStringTable(symbolSection.ReaderAt, f.ByteOrder, readLocation, allocatedBufferForSymbolNameRead)
+		if !ok {
+			continue
+		}
+
+		// Trying to get string representation of symbol.
+		// If the symbol name's length is not in the boundaries [minSymbolNameSize, maxSymbolNameSize+1] then we fail,
+		// and continue to the next symbol.
+		symbolName, ok := getSymbolNameByEntry(f.Sections[symbolSection.Link].ReaderAt, stringEntry, minSymbolNameSize, symbolNameBuf)
+		if !ok {
+			continue
+		}
+
+		// Checking the symbol is relevant for us.
+		if _, ok := copyOfWantedSymbols[symbolName]; !ok {
+			continue
+		}
+
+		// Complete the symbol reading.
+		var symbol elf.Symbol
+		if err := readRestOfSymbol32(&symbol, symbolSection.ReaderAt, f.ByteOrder, symbolName, readLocation+4, allocatedBufferForSymbolRead); err != nil {
+			continue
+		}
+		symbols = append(symbols, symbol)
+		// If relevant, delete it for optimization purposes.
+		delete(copyOfWantedSymbols, symbolName)
+		// If no symbols left, stop running.
+		if len(copyOfWantedSymbols) == 0 {
+			break
+		}
+	}
+
+	return symbols, nil
+}
+
+func getSymbols(f *elf.File, typ elf.SectionType, wanted map[string]struct{}) ([]elf.Symbol, error) {
+	switch f.Class {
+	case elf.ELFCLASS64:
+		return getSymbols64(f, typ, wanted)
+
+	case elf.ELFCLASS32:
+		return getSymbols32(f, typ, wanted)
```

Comment:
you don't really need 2 methods, they are identical except for the `SymSize` const, just pass it as an argument to a unified function